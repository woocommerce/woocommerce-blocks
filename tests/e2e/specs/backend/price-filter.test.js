/**
 * External dependencies
 */
import {
	openDocumentSettingsSidebar,
	switchUserToAdmin,
	getAllBlocks,
	getEditedPostContent,
} from '@wordpress/e2e-test-utils';
import {
	clearAndFillInput,
	visitBlockPage,
} from '@woocommerce/blocks-test-utils';
import { queries, getDocument } from 'pptr-testing-library';

/**
 * Internal dependencies
 */
import {
	insertBlockDontWaitForInsertClose,
} from '../../utils.js';

const block = {
	name: 'Filter Products by Price',
	slug: 'woocommerce/price-filter',
	class: '.wp-block-woocommerce-price-filter',
};

describe( `${ block.name } Block`, () => {

	describe( 'after compatibility notice is dismissed', () => {
		beforeAll( async () => {
			await switchUserToAdmin();
			await visitBlockPage( `${ block.name } Block` );
		} );

		it( 'can only be inserted once', async () => {
			await insertBlockDontWaitForInsertClose( block.name );
			expect( await getAllBlocks() ).toHaveLength( 1 );
		} );

		it( 'renders without crashing', async () => {
			await expect( page ).toRenderBlock( block );
		} );

		it( 'allows title to be manipulated', async () => {
			// we focus on the block
			await page.click( block.class );
			await openDocumentSettingsSidebar();
			await clearAndFillInput(
				'.wp-block[data-type="woocommerce/price-filter"] textarea.wc-block-editor-components-title',
				'New Title'
			);
			await page.click( block.class );
			await page.click(
				'.components-toolbar button[aria-label="Heading 6"]'
			);
			await expect(
				page
			).toMatchElement(
				'.wp-block[data-type="woocommerce/price-filter"] h6 textarea',
				{ text: 'New Title' }
			);

			expect( await getEditedPostContent() ).toMatchSnapshot();
			await clearAndFillInput(
				'.wp-block[data-type="woocommerce/price-filter"] textarea.wc-block-editor-components-title',
				'Filter by price'
			);

			await page.click(
				'.components-toolbar button[aria-label="Heading 3"]'
			);
		} );

		it( 'allows display style to change', async () => {
			await openDocumentSettingsSidebar();

			await page.click( 'button[aria-label="Price Range: Text"]' );
			await expect( page ).toMatchElement(
				'.wc-block-price-filter__range-text'
			);
			await page.click( 'button[aria-label="Price Range: Editable"]' );
		} );

		it( 'allows you to toggle go button', async () => {
			await openDocumentSettingsSidebar();

			const { getByLabelText } = queries;
			const document = await getDocument( page );
			const showGoButton = await getByLabelText( document, /Filter button/i );
			await showGoButton.click();
			await expect( page ).toMatchElement(
				'button.wc-block-filter-submit-button.wc-block-price-filter__button'
			);
			expect( await getEditedPostContent() ).toMatchSnapshot();
			await showGoButton.click();
		} );
	} );
} );
