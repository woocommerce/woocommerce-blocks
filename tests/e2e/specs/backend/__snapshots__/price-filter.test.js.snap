// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter Products by Price Block after compatibility notice is dismissed allows title to be manipulated 1`] = `
"<!-- wp:woocommerce/price-filter {\\"heading\\":\\"New Title\\",\\"headingLevel\\":6} -->
<div class=\\"wp-block-woocommerce-price-filter is-loading\\" data-showinputfields=\\"true\\" data-showfilterbutton=\\"false\\" data-heading=\\"New Title\\" data-heading-level=\\"6\\"><span aria-hidden=\\"true\\" class=\\"wc-block-product-categories__placeholder\\"></span></div>
<!-- /wp:woocommerce/price-filter -->"
`;

exports[`Filter Products by Price Block after compatibility notice is dismissed allows you to toggle go button 1`] = `
"<!-- wp:woocommerce/price-filter {\\"showFilterButton\\":true} -->
<div class=\\"wp-block-woocommerce-price-filter is-loading\\" data-showinputfields=\\"true\\" data-showfilterbutton=\\"true\\" data-heading=\\"Filter by price\\" data-heading-level=\\"3\\"><span aria-hidden=\\"true\\" class=\\"wc-block-product-categories__placeholder\\"></span></div>
<!-- /wp:woocommerce/price-filter -->"
`;
